Source/Particles/Pusher/UpdatePosition.H

```diff
@@ -18,7 +18,7 @@
 /** \brief Push the particle's positions over one timestep,
  *    given the value of its momenta `ux`, `uy`, `uz` */
 AMREX_GPU_HOST_DEVICE AMREX_INLINE
-void UpdatePosition(amrex::ParticleReal& x, amrex::ParticleReal& y, amrex::ParticleReal& z,
```

Comment:
This line doesn't compile when setting `WarpX_PRECISION` to DOUBLE and `WarpX_PARTICLE_PRECISION` to SINGLE. Changing the declaration to be `amrex::ParticleReal` instead of `amrex::ParticleReal&` fixes the compile error, but also breaks tests.